### ISSUE TITLE
Data not working

### ISSUE THREAD

**Issue by Andrufus** [LegacyHuawei/local_manifests#2]
Since I installed the new version for my U8860 (cm-12.1-20150612) I can't manage to make data connexion work at all. In version 12.0 it worked by switching airplane mode on and off but not anymore.
Is there a work around ? Are you aware of this ? or am I just not doing something correctly..

**Comment by rtammekivi** [comment]
I am aware of the bug, however for me the data connection still works. You can also try to turn off data connection and reenable it. Do you happen to know whether it happens in cm-11.0 too?

PS Thanks for using the new issue tracker :)

**Comment by eliotcougar** [comment]
I'm going to test it soon on 20150803 build...
While using the previous build I noticed that it now shows notification "Please log into network" every time in the morning after I disconnect from my home WiFi and go to work... Data connection is unavailable until I manually tap that notification...
It has problems connecting to 3G only if it has been constantly connected to WiFi for many hours (i.e. during the night)... Unlike at home, at work my wifi constantly connects and disconnects while I'm moving around, so that I don't have problems after I leave the building...

**Comment by eliotcougar** [comment]
Well... I left my home and went to work after the phone has been connected to WiFi for the whole night...
The issue is still present, just as I described it in the previous post... No data connection after I disconnect from WiFi, then I have to turn airplane mode on and off, then the notification "Log into network" appears, then I have to tap that, then data connection is established... This is so annoying...
U8860, 20150803 build... At least the performance seems better in this build... Less OOM events...

**Comment by rtammekivi** [comment]
I'm having a hard time debugging it, my signal connection sucks where the PC is at (not to mention it's otherwise very low too). I do remember that it seemed like RIL returned correct SETUP_DATA_CALL information, but the data connection did not work. We are using open source libril, with the only modification of redirecting unsupported commands and changing the SIM I/O commands. It might be possible that the proprietary libril had some modifications.

Do you know if the bug is only present in LegacyHuawei ROMs?

--EDIT--
To add here, there are some binaries that I have removed in my ROM that do some Huawei modem magic:
- callife - binary that is looping once per 6 hours and sends a command to modem
- modempre - binary that seems to handle stock ROM updates, writes EFS to modem and calls reboot if it wrote it (this is why in stock ROM updates the phone reboots while booting).
- oem_rpc_svc - binary that starts the Huawei custom link between modem and ROM, allowing custom Huawei commands to be passed (callife, modempre use this).

**Comment by eliotcougar** [comment]
I think it has been there since the first custom 5.0 ROM I installed... CM11 ROM I had long time ago rebooted itself quite often... I can't be sure if it had this bug or not, it never worked long enough without a reboot...
Additional notification asking me to log into network started appearing in 20150612...
In 20150510 switching airplane mode on and off was enough to fix data link...

UPD: 'callife' sounds to me like some bad Chinese 'keepalive'... :)

**Comment by rtammekivi** [comment]
Yeah, it does seem a bit strange and I will test it out for a while. I also noticed oem_rpc_svc calls modempre itself, so it's more or less an extension to oem_rpc_svc, while being independent. I will prepare a small update package that installs these files and modifies ramdisk so the services are started for testing.

**Comment by rtammekivi** [comment]
Ok, the update package is here: https://basketbuild.com/filedl/devs?dev=LegacyHuawei&dl=LegacyHuawei/tests/b919-modemfiles-050815-u8860.zip

It installs callife, modempre and oem_rpc_svc, also installs customized boot.img over the last release which modifies ramdisk's init.qcom.rc to start callife and oem_rpc_svc.

**Comment by eliotcougar** [comment]
Alright... Downloading, testing... However it'll take a long time since the bug shows itself after being connected to WiFi without interruption for more than 6 (???) hours...
~20 hours later, tomorrow, I'll give a report...

Small update: Not sure if it's this update package or the 20150803 build itself, but the dreaded "long delay between incoming call and phone interface appearing" is finally almost gone... The buttons to answer the call appear much sooner now...

**Comment by eliotcougar** [comment]
Good morning (Yes, 14:00 is morning)...
Unfortunately, this annoying bug is still present after the patch... No difference at all, including the "Log into network" notification... :(

**Comment by rtammekivi** [comment]
I realized that the callife binary sleeps for 21600 seconds, but this does not probably account device suspend time. It could be something totally different, but it's weird to have this kind of binary then. What would be counted after 6 hours of device CPU on time?

Anyway, I guess I could try to use closed source libril with frameworks modifications to see if it has any effect. Who knows, maybe 3.4 kernel is incompatible in some way...

I don't know why you have the "Log into network" notification though - maybe some checkbox is off in the APN settings?

**Comment by JekaPinsk** [comment]
I wonder if it could be used with 404020 baseband.

**Comment by eliotcougar** [comment]
I just figured out that I don't have to tap "Log into network" notification... It actually establishes 3G connection after a while, and the notification disappears...

**Comment by rtammekivi** [comment]
Apparently the issue seems to be fixed for me, at least I can no longer replicate it. Have you still had the issue?

**Comment by eliotcougar** [comment]
The issue described in the first post seems to be a little different...
I still don't have 3G connection every morning after I go away from my WiFi hotspot after being connected to it for the whole night... :(

**Comment by bibilemihai** [comment]
Hi I have not seen the bug tracking on git :) so I've already posted my experience on XDA.
For me none of your roms are working (no data at all) & I was wondering if it is not linked to the baseband 404029... 

However the data is there if I install any other CM (not yours unfortunatelly)

any idea ?
